### PR TITLE
cpp backend for constant dictionary

### PR DESCRIPTION
**Pull request by Smit-create**

### CODE REVIEW

File: src/libasr/codegen/asr_to_cpp.cpp [hunk 1]
```diff
@@ -593,6 +593,21 @@ Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
         last_binary_plus = false;
     }
 
+    void visit_ConstantDictionary(const ASR::ConstantDictionary_t &x) {
+        LFORTRAN_ASSERT(x.n_keys == x.n_values);
```
Comment (namannimmo10) [inline src/libasr/codegen/asr_to_cpp.cpp:597]: I don't think we need the check here since we already do this in AST->ASR.

Comment (certik) [reply src/libasr/codegen/asr_to_cpp.cpp:597]: Yes, we don't need to check, but it doesn't hurt either. The assert statements have no overhead in Release mode.

**Comment by certik** [comment]
And add tests that show the newly implemented C++ output.

**Review by certik — APPROVED** [review]

**Comment by Smit-create** [comment]
Thanks for the review!

**Pull request merged**